scripts/process_output.py

```diff
@@ -308,28 +309,40 @@ def process_metadata(ds):
     """
     s = json.dumps(ds.attrs, cls=NumpyEncoder)
     # Hash based on the metadata of the run
-    run_id = hashlib.sha256(s.encode("utf-8")).hexdigest()[:8]
-    newpath = path / f"pv-ready-data-{run_id}"
-    # Make the new directory if it doesn't exist
+    hash_digest = hashlib.sha256(s.encode("utf-8")).hexdigest()[:8]    
+    ds = ds.assign_attrs(hash_digest=hash_digest) 
+    # Determine the institute based on the project name
+    project = ds.attrs.get("project", "")
+    if project.startswith("a8b1"):
+        institute = "SWPC"
+    elif project.startswith("ENLIL."):
+        institute = "CCMC"
+    elif project.startswith("/data"):
+        institute = "SWxTREC"
+    else:
+        institute = "UNKNOWN"
+        warnings.warn("Unknown institute, defaulting to UNKNOWN")
+    ds = ds.assign_attrs(institute=institute)
+
+    # Extract the run_id from the dataset attributes
+    if not run_id:
+        # In order to extract the run_id from a SWPC run, the path must end with wsa_enlil_57484.57285344.dbqs0
+        if institute == "SWPC" and re.match(r"^.*wsa_enlil_\d{5}\.\d*\.dbqs0$", path.name):
```

Comment:
Where does "path" come from here...? This seems like it is coming from outside of this scope, so is this a global or is there a way you can pass this path in instead of newpath?